src/HomeScreen/index.jsx

```diff
@@ -10,24 +10,34 @@ import {
   colors,
   useKeepAwake,
   BaniList,
+  defaultBaniOrder,
 } from "@common";
 import styles from "./styles";
 import BaniHeader from "./components/BaniHeader";
-import { useAppFirstTime, useBaniLength, useBaniList } from "./hooks";
+import { useBaniLength, useBaniList } from "./hooks";
+import errorHandler from "../common/errHandler";
+import { setBaniOrder } from "../common/actions";
 
 const HomeScreen = React.memo(({ navigation }) => {
+  const [error, setError] = useState(null);
   const { navigate } = navigation;
-  const { baniListData } = useBaniList();
+  const { baniListData, fetchBaniList } = useBaniList(setError);
   const isNightMode = useSelector((state) => state.isNightMode);
   const isStatusBar = useSelector((state) => state.isStatusBar);
   const language = useSelector((state) => state.language);
   const theme = useSelector((state) => state.theme);
+  const baniOrder = useSelector((state) => state.baniOrder);
+
   useKeepAwake();
   useScreenAnalytics(constant.HOME_SCREEN);
-  const isAppOpenFirstTime = useAppFirstTime();
   const { baniLengthSelector } = useBaniLength();
   const dispatch = useDispatch();
 
+  if (error) {
```

Comment:
Shouldn't this be inside a useEffect, as it's based on a state change?

Reply:
Fixed